File: reservation-service/domain/reservation.go
package domain

import (
	"encoding/json"
	"io"
	"time"

	"github.com/gocql/gocql"
)

type Reservation struct {
	ReservationID gocql.UUID `json:"reservationId"`
	Location      string     `json:"location"`
	AccommID      string     `json:"accommId"`
	GuestEmail    string     `json:"guestEmail"`
	HostEmail     string     `json:"hostEmail"`
	Price         int        `json:"price"`
	NumOfPeople   int        `json:"numOfPeople"`
	StartDate     time.Time  `json:"startDate"`
	EndDate       time.Time  `json:"endDate"`
}

type Reservations []*Reservation

func (a *Reservations) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(a)
}

func (a *Reservation) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(a)
}

func (a *Reservation) FromJSON(r io.Reader) error {
	d := json.NewDecoder(r)
	return d.Decode(a)
}

File: front-end-service/src/app/model/reservation.ts
import { formatDate } from "@angular/common";

export class Reservation{
    reservationId: string;
    accommodationId: string;
    location: string;
    guestEmail: string;
    hostEmail: string;
    price: number;
    numOfPeople: number;
    startDate: Date;
    endDate: Date;
    constructor() {
        this.reservationId = ""
        this.accommodationId = ""
        this.location = ""
        this.guestEmail = ""
        this.hostEmail = ""
        this.price = 0
        this.numOfPeople = 0
        this.startDate = new Date(formatDate(Date.now(),'yyyy-MM-dd','en_us'))
        this.endDate = new Date(formatDate(Date.now(),'yyyy-MM-dd','en_us'))
    }
}